release-notes/Cabal-3.14.0.0.md

```diff
@@ -157,6 +155,58 @@ Cabal and Cabal-syntax 3.14.0.0 changelog and release notes
   configure"). This has recently caught out even Cabal devs. Clarify these
   messages. [#9476](https://github.com/haskell/cabal/pull/9476)
 
+- In [#9718](https://github.com/haskell/cabal/pull/9718) constructors
+  `PackageDB`: `GlobalPackageDB`, `UserPackageDB`, `SpecificPackageDB` have
+  been moved to an auxiliary datatatype `PackageDBX`.
+
+  Change imports:
+
+    ```haskell
+    import Distribution.Simple.Compiler
+       (PackageDB, PackageDBX (GlobalPackageDB,
+                               UserPackageDB,
+                               SpecificPackageDB))
+    ```
+
+  - **Fields of `ConfigFlags`, `BuildFlags`, `InstallFlags`, `TestFlags`,
+    `BenchmarkFlags`, `HaddockFlags`, `HscolourFlags`, `SDistFlags`,
+    `CopyFlags`, `RegisterFlags`, `CleanFlags`, `ReplFlags`**
+
+    These fields have partially moved to `CommonFlags`. Use corresponding
+    pattern synonyms and/or Monoid instance of `CommonSetupFlags`.
+
+    Example for haddock command:
+
+    ```diff
+     import Distribution.Simple.Setup (HaddockFlags(..))
+    +import Distribution.Simple.Setup.Common (CommonSetupFlags(..))
+
+     example =
+       someHaddockFlags
+    -    { haddockVerbosity = a
+    -    , haddockTargets  = b
+    +    { haddockCommonFlags = mempty
+    +      { setupVerbosity = a
+    +      , setupTargets = b
+    +      }
+         }
+    ```
+
+  - **Additions to `SymbolicPath`, `RelativePath`**
+
+    Module
+    [Distribution.Utils.Path](https://hackage.haskell.org/package/Cabal-syntax-3.14.0.0/docs/Distribution-Utils-Path.html),
+    now provides more nuanced API that Cabal uses to keep track of filepath
+    locations. (Hopefully, avoiding confusion which things should go where and
+    how.)
+
+    In your specific circumstance, you'll need to decide how much of that nuance
```

Comment:
I think it should be pointed out that using `getSymbolicPath` has some nuance.

* FilePath in `cabal-install` is always relative to the project directory which is where cabal-install is run.
* SymbolicPath is relative to the package root of the particular package `Cabal` is compiling at that time. 

For a `./Setup.hs` you are probably fine using `interpretSymbolicPathCWD`, but that is only because `cabal-install` calls `./Setup` in the package directory. If you are writing an application using `Cabal` library, you might need to take the working directory into account propertly.

Reply:
Yep, that basically echoes what the haddocks (+types) say.

I didn't want to overload release notes with deep details; so there's no "do this" code sample, but the call to action is to go check the updated module reference.

Reply:
You two have more experience than me with these modules. The warning (“[…] but see the linked module's haddocks for caveats and less drastic options”) seems assertive enough.

Any modification proposal is welcome!